plugins/modem-manager/fu-mm-utils.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2019 Aleksander Morgado <aleksander@aleksander.es>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include <gio/gio.h>
+#include "fu-mm-utils.h"
+
+gboolean
+fu_mm_utils_get_udev_port_info (GUdevDevice  *device,
+				gchar	    **out_device_sysfs_path,
+				gint	     *out_port_ifnum,
+				GError	    **error)
+{
+	g_autoptr(GUdevDevice) parent = NULL;
+	g_autofree gchar *device_sysfs_path = NULL;
+	gint port_ifnum = -1;
+	const gchar *aux;
+
+	/* ID_USB_INTERFACE_NUM is set on the port device itself */
+	aux = g_udev_device_get_property (device, "ID_USB_INTERFACE_NUM");
+	if (aux != NULL)
+		port_ifnum = (guint16) g_ascii_strtoull (aux, NULL, 16);
```

Comment:
I guess g_udev_device_get_property_as_uint64() is using a hardcoded base 10, right?

Reply:
It tries to guess in the same way as strtoll() does, but in our case the ID_USB_INTERFACE_NUM is always in hex and without a "0x" prefix, so we do need to explicitly use base 16